bin/rna_seq.py

```diff
@@ -1,67 +1,82 @@
 #!/usr/bin/env python3
 
-import sys
-import ast
-import requests
+import query_ena_transcriptome
+import logging
+import argparse
 import csv
-import os
-import subprocess
-from datetime import datetime
-import my_process as mp
-import csv_ENA_download
 
-def generate_ena_csv(tax_ranks,genome ,baseDir):
-    data_found = False
-    for l in range(0,3):
-        current_name = "level_" + str(l) + "_name"
-        current_tax = "level_" + str(l) + "_tax"
-        current_hierarchy = "level_" + str(l) + "_hierarchy"
-        user_prefered_tax = False
-        #TO DO - review filtering conditions, otherwise ranks_dict["species"] is just enough.
-        #if we haven't found transcriptomic data at species level, don't go further and return null.
-        if tax_ranks[current_hierarchy] < mp.ranks_dict["species"] and not data_found:
-            g_name = tax_ranks[current_name]
+logging.basicConfig(level=logging.INFO)
+
+#check if we need to include thse after sourmash
+#MAX_NB_SAME_DESCRIPTORS = 6
+#MAX_NB_TOTAL_FILES = 3
+#check which descriptor is used - sample name in this case
+#check if filtering of descripions for RNAs is needed
 
-            try:
-                genome_tax = tax_ranks["rna_acc"]
-                user_prefered_tax = True
-            except:
-                genome_tax = tax_ranks[current_tax]
+def parse_taxa(taxa_file):
+    logging.info("Parsing taxa lineages from file")
+    tax_dict = {}
+    with open(taxa_file, 'r') as taxa:
+        for line in taxa:
+            data = line.rstrip().split('\t')
+            #   rank: taxid
+            tax_dict[data[1]] = data[2]
+    return tax_dict
 
-            output_rna_csv_path = genome + "_" + str(genome_tax) + "_ENA_rna.csv"
-            log_dir = os.path.join(baseDir, 'logs', genome)
-            log_file_path = log_dir + '/rna_script.log'
+def find_transcriptome_data(tax_ranks, preferred_rank = False):
+    """use preferred rank if given by user. Circle through taxa until transcriptome data is found"""
+    if preferred_rank:
+        taxid = tax_ranks[preferred_rank]
+        ena = query_ena_transcriptome.EnaMetadata(taxid)
+        if len(transciptome_metadata):
```

Comment:
change 
"transciptome_metadata" to 
"transcriptome_metadata" 
----------------------------
I am not sure I understand this, is the first "if len(transciptome_metadata):" supposed to return always false? I don't see "transciptome_metadata" defined/initialised for predefined ranks.

Reply:
Yep thanks for spotting this. I just spotted it today as well running at scale. The loop just after wasn't quite right as well. I'll let the pipeline finish and then resolve this comment.